node/src/chain_spec/cs_staking.rs

```diff
@@ -0,0 +1,224 @@
+        babe: BabeConfig {
+            authorities: vec![],
+            epoch_config: Some(main_runtime::constants::BABE_GENESIS_EPOCH_CONFIG),
+        },
+        im_online: ImOnlineConfig { keys: vec![] },
+        authority_discovery: AuthorityDiscoveryConfig { keys: vec![] },
+        grandpa: GrandpaConfig {
+            authorities: vec![],
+        },
+        staking: StakingConfig {
+            stakers: initial_authorities
+                .iter()
+                .map(|x| (x.0.clone(), None, STASH))
+                .collect(),
+            invulnerables: initial_authorities.iter().map(|x| x.0.clone()).collect(),
+            ..Default::default()
+        },
+
+        // Governance
+        // Root Committee
+        root_committee: Default::default(),
+        root_membership: RootMembershipConfig {
+            members: roots.clone(),
+            phantom: Default::default(),
+        },
+    }
+}
+
+fn local_staking_genesis() -> GenesisConfig {
+    testnet_genesis(
+        vec![
+            get_authority_keys_from_seed("Alice"),
+            get_authority_keys_from_seed("Bob"),
+        ],
+        vec![
+            get_account_id_from_seed::<sr25519::Public>("Alice"),
+            get_account_id_from_seed::<sr25519::Public>("Bob"),
+            get_account_id_from_seed::<sr25519::Public>("Charlie"),
+        ],
+        vec![get_account_id_from_seed::<sr25519::Public>("Ferdie")],
+        None,
+    )
+}
+
+/// Local testnet config to test the staking pallet
+pub fn local_staking_config() -> ChainSpec {
+    ChainSpec::from_genesis(
+        "Local Staking Testnet",
+        "local_staking_testnet",
+        ChainType::Local,
+        local_staking_genesis,
+        vec![],
+        None,
+        Some("nodl"),
+        Some(build_local_properties()),
+        Default::default(),
+    )
+}
+
+fn development_config_genesis() -> GenesisConfig {
+    testnet_genesis(
+        vec![get_authority_keys_from_seed("Alice")],
+        vec![
+            get_account_id_from_seed::<sr25519::Public>("Alice"),
+            get_account_id_from_seed::<sr25519::Public>("Bob"),
+            get_account_id_from_seed::<sr25519::Public>("Charlie"),
+            get_account_id_from_seed::<sr25519::Public>("Dave"),
+        ],
+        vec![get_account_id_from_seed::<sr25519::Public>("Ferdie")],
+        None,
+    )
+}
+
+/// Development config (single validator Alice)
+pub fn development_config() -> ChainSpec {
+    ChainSpec::from_genesis(
+        "Development Staking Testnet",
+        "dev_staking_testnet",
+        ChainType::Development,
+        development_config_genesis,
+        vec![],
+        None,
+        Some("nodl"),
+        Some(build_local_properties()),
+        Default::default(),
+    )
+}
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::*;
+    use sp_runtime::BuildStorage;
+
+    #[test]
+    fn test_create_staking_local_chain_spec() {
+        local_staking_config().build_storage().unwrap();
+    }
+}
```

Comment:
we should test the dev config if it is used too

Reply:
My Bad, added now. Done :+1: